zookeeper-docs/src/main/resources/markdown/zookeeperProgrammers.md

```diff
@@ -503,6 +503,85 @@ new list and failing to connect, the client moves back to the normal mode of ope
 an arbitrary server from the connectString and attempts to connect to it. If that fails, it will continue
 trying different random servers in round robin. (see above the algorithm used to initially choose a server)
 
+**local session**.Added in 3.5.0, mainly implemented by [ZOOKEEPER-1147](https://issues.apache.org/jira/browse/ZOOKEEPER-1147).
+
+- Background:The create and close of session is costly in zookeeper, especially when these operations of sessions need a
+quorum confirmation by leader, which we call this session type: global session. it will be a bottle neck for ZooKeeper ensemble
+when it wants to handle thousands of client connections.
```

Comment:
The creation and closing of sessions are costly in ZooKeeper because they need quorum confirmations. They become the bottleneck of a ZooKeeper ensemble when it needs to handle thousands of client connections.